src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -130,13 +130,17 @@ def get_first_week_from_profile(self, profile_uuid, current_timestamp) -> dict:
             and datapoint.time <= first_datapoint_time + duration(days=7)
         )
 
-        datapoint_dict = {
-            self._strip_timezone_and_create_pendulum_instance_from_datetime(datapoint.time).set(
-                year=current_timestamp.year,
-                month=current_timestamp.month,
-                day=current_timestamp.day
-            ): datapoint.value for datapoint in datapoints
-        }
+        datapoint_dict = {}
+        diff_current_to_db_time = None
+        for datapoint in datapoints:
+            db_time = self._strip_timezone_and_create_pendulum_instance_from_datetime(
+                datapoint.time)
+            if diff_current_to_db_time is None:
```

Comment:
I'd suggest to move this before the `for` loop as it's calculated only once with the first element.

Reply:
I Ieft a verbose comment one line below in order to explain why it is here. Let me know if it is not clear.

Reply:
Yes. I do get it. However, you may be a able to evaluate the query before the for loop.
```python
datapoints = list(datapoints)
```
I guess this way you are not refetching the whole set from the DB again.

Reply:
Good idea, will change this!

Reply:
Done [890adb1](https://github.com/gridsingularity/gsy-e/pull/1603/commits/890adb1115f6135eba8843a91a4667ecb38c187e)